Kernel/Syscalls/sigaction.cpp

```diff
@@ -73,6 +74,46 @@ KResultOr<FlatPtr> Process::sys$sigaction(int signum, Userspace<const sigaction*
     return 0;
 }
 
+KResultOr<FlatPtr> Process::sys$sigaltstack(Userspace<const stack_t*> ss, Userspace<stack_t*> old_ss)
+{
+    VERIFY_PROCESS_BIG_LOCK_ACQUIRED(this)
+    REQUIRE_PROMISE(sigaction);
+
+    stack_t old_ss_value;
+    old_ss_value.ss_sp = (void*)Thread::current()->m_alternative_signal_stack;
+    old_ss_value.ss_size = Thread::current()->m_alternative_signal_stack_size;
+    old_ss_value.ss_flags = 0;
+    if (!Thread::current()->has_alternative_signal_stack())
+        old_ss_value.ss_flags = SS_DISABLE;
+    else if (Thread::current()->is_in_alternative_signal_stack())
+        old_ss_value.ss_flags = SS_ONSTACK;
+
+    if (ss) {
+        stack_t ss_value;
+        TRY(copy_from_user(&ss_value, ss));
+
+        if (Thread::current()->is_in_alternative_signal_stack())
+            return EPERM;
+
+        if (ss_value.ss_flags == SS_DISABLE) {
+            Thread::current()->m_alternative_signal_stack_size = 0;
+            Thread::current()->m_alternative_signal_stack = 0;
+        } else if (ss_value.ss_flags == 0) {
+            if (ss_value.ss_size < MINSIGSTKSZ)
+                return ENOMEM;
+            Thread::current()->m_alternative_signal_stack = (FlatPtr)ss_value.ss_sp;
```

Comment:
Shouldn't this have some sort of locking?

Reply:
We're holding the big process lock, so I assume we don't need to?